app/lib/activitypub/activity/like.rb

```diff
@@ -17,16 +17,16 @@ def perform
 
   # Misskey delivers reactions as likes with the emoji in _misskey_reaction
   # see https://misskey-hub.net/ns.html#misskey-reaction for details
-  def maybe_process_misskey_reaction(original_status)
+  def maybe_process_misskey_reaction
+    original_status = status_from_uri(object_uri)
     name = @json['_misskey_reaction']
     return false if name.nil?
 
     custom_emoji = nil
     if /^:.*:$/.match?(name)
```

Comment:
This is a repeating pattern and might be worth creating a method.
I'm not sure where to put such a method though.